Feat/ui-customisations

amjedidiah opened the pull request:
# Upup Pull Request Template

## Description

-   [x] Container: Background, size and border radius
-   [x] Progress bar & percentage style
-   [x] Files & previews container (grid, flex,...)
-   [x] Providers list style
-   [x] Each button style & colour individually: including remove file button

AND more

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings

amjedidiah committed:
fix(app): preview fixes

- fixed bug with preview
- properly managed preview and file click event handlers
- icon support for more file extensions
- deleted unused provider file

amjedidiah committed:
refactor(app): Rename "Provider" to "upupProvider" to avoid confusion with"react Provider"

amjedidiah committed:
fix(app): fixed upload progress and percentage

amjedidiah committed:
fix(app): s3 upload

- properly managed conditions for different s3 uploaders
- returned public url for uploaded image in response to client

amjedidiah committed:
fix(app): unique files

- fix for files being selected multiple times

amjedidiah committed:
fix(app): upload key

- removing `uploads` from the file upload key

amjedidiah committed:
refactor(app): app refactor

- refactored app to minimise code repetitions

#### Fixes
-   onFileSelected fix
-   Fix for same file selected multiple times
-   Fix for Google Drive loading

amjedidiah committed:
feat(app): started ui design

amjedidiah committed:
feat(app): MainBox component
- reverting back to using a single box to manage preview and adapter options

amjedidiah committed:
feat(ui): adapter view improvements

-   [x] Cancel button to be visible when adding More files
-   [x] Show icon of drive instead of name
-   [x] Replace user name with user image/icon in drive selection
-   [x] Apply background color same as Preview component in drive selection
-   [x] Search functionality for drive files

amjedidiah committed:
feat(ui): Error display on adding files

amjedidiah committed:
feat(ui): UpUp mini

amjedidiah committed:
feat(ui): more ui improvements

- [x] Progress bar  for single uploads
- [x] Progress bar and progress value for all uploads
- [x] Sonner toast as default error handler instead of `console.error`
- [x] minor design fixes

amjedidiah committed:
fix(app): review fixes

-   [x] allProgress: Math.round
-   [x] individual progress bar: rounded bottom only
-   [x] show warning for existing file selected again
-   [x] preview thumbnail: available for supported types
-   [x] preview: only show for supported file types
-   [x] disable all buttons during upload
-   [x] Uploads done with errors should not show Done button

amjedidiah committed:
feat(ui): dark mode

amjedidiah committed:
docs(app): better storybook doc

- single storybook story with customizable controls
- minor design changes

amjedidiah committed:
feat(ui): ui customisations

-   [x] Container: Background, size and border radius
-   [x] Progress bar & percentage style
-   [x] Files & previews container (grid, flex,...)
-   [x] Providers list style
-   [x] Each button style & colour individually: including remove file button

AND more

amjedidiah committed:
fix(ui): fix for classnames not applying

amjedidiah committed:
chore(tools): eslint setup

amjedidiah committed:
fix(ui): preview fixes

- more ui customizations
- improved overall background customisation support
- file preview fix
- lint fixes
- improved camera image preview
- fix for starting file limit in Storybook
- better file progress tracking and style

amjedidiah committed:
feat(ui): responsive ui with container rules

amjedidiah committed:
fix(prettier): prettier fixes

amjedidiah committed:
feat(app): unique file by url + prettier fixes

amjedidiah committed:
feat(ui): improved drive styling

amjedidiah committed:
fix(preview): better pdf preview handling

amjedidiah committed:
fix(review): review fixes

- max z-index for powered by upup
- file preview fix
- proper preview for storybook component

amjedidiah committed:
fix(ui): file preview portal z-index

BSalaeddin reviewed (commented):
Looks good to me @amjedidiah

amjedidiah committed:
refactor(app): docs updates

- renamed `adapterList` to `adapterButtonList` and `adapterDriveLoading` to `driveLoading` to match naming convention
- moved `loader` props into `icon` props
- improved `validateConfig` functionality
- updated README with `Getting Started` docs from landing page

amjedidiah committed:
fix(publish): npm publish fix

MedAmine1212 reviewed (commented):
Fix prettier issues & merge @amjedidiah

amjedidiah committed:
fix(app): prettier fixes
